src/AwsS3Adapter.php

```diff
@@ -438,7 +448,7 @@ public function listContents($dirname = '', $recursive = false)
      */
     protected function normalizeResponse(array $object, $path = null)
     {
-        $result = ['path' => $path ?: $this->removePathPrefix($object['Key'])];
+        $result = ['path' => $path ?: $this->removePathPrefix(isset($object['Key']) ? $object['Key'] : $object['Prefix'])];
```

Comment:
And the best comment on a PR award goes tooooooooo ... @zaak! Seriously, thanks for this, this is exactly what I wanted to know.